### PR TITLE
Build: initialize base trivy go plugin

### PR DESCRIPTION
**Pull request by malikovmm**
This pull request is a continuation https://github.com/afdesk/trivy-go-plugin/pull/1

### CODE REVIEW

**Review by afdesk — CHANGES_REQUESTED** [review]

File: main.go [hunk 1]
```diff
@@ -1,41 +1,158 @@
...
 	}
 }
+
 func closeFile(file *os.File) {
 	if err := file.Close(); err != nil {
 		log.Fatalf("failed to remove file %v", err)
 	}
 }
+
+func getFlagValue(flag string) *string {
```
[the remark this answers is not in the document]
Comment (malikovmm) [reply main.go:107]: the return value can be nil

File: main.go [hunk 2]
```diff
@@ -1,41 +1,157 @@
...
 		log.Fatalf("failed to remove file %v", err)
 	}
 }
+
+func getFlagValue(flag string) string {
+	flagIndex := slices.Index(os.Args, flag)
+	if flagIndex != -1 && (len(os.Args)-1) > flagIndex { // the flag exists and it is not the last argument
+		flagValue := os.Args[flagIndex+1]
+		return flagValue
+	}
```
Comment (afdesk) [inline main.go:113]:
```suggestion
	flagIndex := slices.Index(os.Args, flag)
	if flagIndex != -1 && (len(os.Args)-1) > flagIndex { // the flag exists and it is not the last argument
		return os.Args[flagIndex+1]
	}
```

File: main.go [hunk 3]
```diff
@@ -1,41 +1,157 @@
...
+		log.Fatal("flag --plugin-output is required")
+	}
+	var trivyOutputFileName string
+	if outputFlagValue := getFlagValue("--output"); outputFlagValue == "" {
+		tempFileName := filepath.Join(os.TempDir(), tempJsonFileName)
+		trivyOutputFileName = tempFileName
+		defer removeFile(trivyOutputFileName)
+	} else {
+		trivyOutputFileName = outputFlagValue
+	}
```
Comment (afdesk) [inline main.go:37]:
```suggestion
	trivyOutputFileName := getFlagValue("--output")
	if  trivyOutputFileName == "" {
		trivyOutputFileName = filepath.Join(os.TempDir(), tempJsonFileName)
		defer removeFile(trivyOutputFileName)
	}
```

**Review by afdesk — CHANGES_REQUESTED** [review]

**Review by afdesk — APPROVED** [review]

**Comment by afdesk** [comment]
thanks

**Pull request merged**